toolchain/parse/node_kind.h

```diff
@@ -47,6 +47,15 @@ CARBON_DEFINE_RAW_ENUM_CLASS(NodeKind, uint8_t) {
 
 // A class wrapping an enumeration of the different kinds of nodes in the parse
 // tree.
+//
+// In order to allow the children of a node to be determined without relying on
+// the subtree size field in the parse node, each node kind must have one of:
+//
+// - a bracketing node kind, which is always the kind for the first child, and
+//   is never the kind of any other child, or
+// - a fixed child count,
+//
+// or both. This is required even for nodes that are invalid.
```

Comment:
```suggestion
// or both. This is required even for nodes produced by an invalid parse.
```

Reply:
```suggestion
// or both. This is required even for nodes that have errors.
```
I specifically mean nodes marked with the `has_errors` flag, so how about this?